brainlit/algorithms/generate_fragments/spline_fxns.py

```diff
@@ -0,0 +1,296 @@
+import operator
+from scipy.interpolate import BSpline, splev
+import numpy as np
+from typing import Iterable, Union, Tuple
+from brainlit.utils.util import (
+    check_type,
+    check_size,
+    check_precomputed,
+    check_iterable_type,
+    check_iterable_nonnegative,
+)
+
+
+def speed(
+    x: np.ndarray,
+    t: np.ndarray,
+    c: np.ndarray,
+    k: np.integer,
+    aux_outputs: bool = False,
+) -> np.ndarray:
+    """Compute the speed of a B-Spline.
+
+    The speed is the norm of the first derivative of the B-Spline
```

Comment:
Reference where you found this equation

Reply:
I found it here [[1](http://math.mit.edu/~varul/F17-18.02/Recitation-5-Solutions.pdf)], [[2](https://www.math.ucdavis.edu/~kouba/Math21BHWDIRECTORY/ArcLength.pdf)], do you want me to refer these in the code too?